src/Core/Translation/Storage/Provider/Definition/CoreDomainProviderDefinition.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Translation\Storage\Provider\Definition;
+
+/**
+ * Properties container for core translation provider filtering by a single domain name.
+ */
+class CoreDomainProviderDefinition extends AbstractCoreProviderDefinition
+{
+    /**
+     * @var string
+     */
+    private $domainName;
+
+    /**
+     * @param string $domainName
+     */
+    public function __construct(string $domainName)
+    {
+        $this->domainName = $domainName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType(): string
+    {
+        return ProviderDefinitionInterface::TYPE_CORE_DOMAIN;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDomainName(): string
+    {
+        return $this->domainName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilenameFilters(): array
+    {
+        return ['#^' . preg_quote($this->domainName, '#') . '([A-Za-z]|\.|$)#'];
```

Comment:
Could you move regex to constant, please ?

Reply:
there is a dynamic element in the regex. Not sure we can put it in a constant

Reply:
Use sprintf with a constant